### PR TITLE
Ported Code to support Python3 and made other subtle changes

### PR DESCRIPTION
**Pull request by akhildevelops**
Issue: https://github.com/xychang/RecursiveHierarchicalClustering/issues/6

- Ported Code to Python3
- Made subtle Changes such as args for cPickle reads and writes.

Tested and Checked for recursiveHierarchicalClustering.py

### CODE REVIEW

File: recursiveHierarchicalClustering.py [hunk 1]
```diff
@@ -946,6 +948,12 @@ def run(sid_seq, outPath, matrixPath, sids, nPath):
     sizeThreshold = 0.05
     return runDiana(outPath, sid_seq, None, matrixPath, sorted(sids))
 
+
+def create_output_dir(path):
+    if not os.path.exists(path):
+        os.makedirs(path)
+    return True
```
Comment (xychang) [inline recursiveHierarchicalClustering.py:955]: Love it :D

File: recursiveHierarchicalClusteringFast.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ def getDia(self, cid, cluster):
                 maxDist = curMax
         return maxDist
 
-    def splitCluster(self, (baseCluster, diameter, cid)):
+    def splitCluster(self, xxx_todo_changeme):
```
Comment (xychang) [inline recursiveHierarchicalClusteringFast.py]: maybe change to cluster_info

File: recursiveHierarchicalClusteringFast.py [hunk 3]
```diff
@@ -102,6 +102,7 @@ def splitCluster(self, (baseCluster, diameter, cid)):
         :rtype sumAB: float
             - sum of all pairwise distance between clusterA and clusterB
         """
+        (baseCluster, diameter, cid) = xxx_todo_changeme
```
Comment (xychang) [inline recursiveHierarchicalClusteringFast.py]: maybe change to cluster_info

File: recursiveHierarchicalClusteringFast.py (hunk 2 above)
Comment (xychang) [inline recursiveHierarchicalClusteringFast.py]: maybe change to cluster_info

File: recursiveHierarchicalClusteringFast.py [hunk 4]
```diff
@@ -187,7 +188,7 @@ def evaluateModularity(self, clusters, sumEntries):
             secondEntry += np.sum(sumRowCur) ** 2
         return (firstEntry - secondEntry / m) / m
 
-    def evaluateModularityShift(self, (clusterA, clusterB), sumAB):
+    def evaluateModularityShift(self, xxx_todo_changeme1, sumAB):
```
Comment (xychang) [inline recursiveHierarchicalClusteringFast.py]: just `clusterAB` is fine

File: recursiveHierarchicalClusteringFast.py [hunk 5]
```diff
@@ -200,6 +201,7 @@ def evaluateModularityShift(self, (clusterA, clusterB), sumAB):
             - sum of all pairwise distance between clusterA and clusterB
         :rtype: float
         """
+        (clusterA, clusterB) = xxx_todo_changeme1
```
Comment (xychang) [inline recursiveHierarchicalClusteringFast.py]: ditto

**Review by xychang — CHANGES_REQUESTED** [review]
Overall looks good to me. Thank you so much for the effort. Would you mind comparing the results before and after the change, just to make sure everything is the same? Thanks!

**Comment by akhildevelops** [comment]
K sure, I'll make the changes and will revert after examining the results, shortly.

**Comment by akhildevelops** [comment]
Hi @xychang ,

Updated change requests. Also checked the results and are same. Even checked the recursiveHierarchicalClusteringFast.py file. Below are the cluster images.

- Updated recursiveHierarchicalClustering.py (python 3)
Cluster_updated

- Updated recursiveHierarchicalClusteringFast.py (python 3)
Fast


- Original recursiveHierarchicalClustering.py (python 2)
Original

**Pull request merged**